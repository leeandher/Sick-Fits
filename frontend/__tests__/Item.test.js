import ItemComponent from "../components/Item"
import { shallow } from "enzyme"
import toJSON from "enzyme-to-json"

const itemProps = {
  id: "ABC123",
  title: "Sick Item",
  price: 4000,
  description: "A super sick item for super sick people!",
  image: "sick.jpg",
  largeImage: "large-sick.jpg"
}

describe("<Item />", () => {
  // A thorough shallow rendering test...

  // const wrapper = shallow(<ItemComponent item={itemProps} />)
  // it("renders the PriceTag and Title properly", () => {
  //   const PriceTag = wrapper.find("PriceTag")
  //   expect(PriceTag.children().text()).toBe("$50")
  //   expect(wrapper.find("Title a").text()).toBe(itemProps.title)
  // })
  // it("renders the image properly", () => {
  //   const img = wrapper.find("img")
  //   expect(img.props().src).toBe(itemProps.image)
  //   expect(img.props().alt).toBe(itemProps.title)
  // })
  // it("renders the buttons properly", () => {
  //   const buttonList = wrapper.find(".buttonList")
  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find("Link").exists()).toBe(true)
  //   expect(buttonList.find("AddToCart").exists()).toBe(true)
  //   expect(buttonList.find("DeleteItem").exists()).toBe(true)
  // })

  it("renders", () => {
    shallow(<ItemComponent item={itemProps} />)
  })
  it("matches the snapshot", () => {
    const wrapper = shallow(<ItemComponent item={itemProps} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
