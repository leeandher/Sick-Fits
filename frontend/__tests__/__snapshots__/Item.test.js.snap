// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> matches the snapshot 1`] = `
<ItemStyles__Item>
  <PriceTag>
    $40
  </PriceTag>
  <Link
    href={
      Object {
        "pathname": "item",
        "query": Object {
          "id": "ABC123",
        },
      }
    }
  >
    <a>
      <img
        alt="Sick Item"
        src="sick.jpg"
      />
    </a>
  </Link>
  <Title>
    <Link
      href={
        Object {
          "pathname": "item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Sick Item
      </a>
    </Link>
  </Title>
  <p>
    A super sick item for super sick people!
  </p>
  <User>
    <Component />
  </User>
</ItemStyles__Item>
`;
